aerogear/mobile-security-service-server
# Added a custom middleware wrapper and added CORS config

## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-8388

## What

Created a custom middleware wrapper where we can use default middlewares and define our own.

## Why

We need a place to set, create and customize middleware. CORS is the first one that we are sure we will use.

## How

1. Added CORS config properties to config file.
2. Created a new package and file in `pkg/middlewares` called `middlewares.go`.
3. Referenced [Echo documentation](https://echo.labstack.com/middleware/cors) to "use" CORS middleware.

## Verification Steps
 
1. Update your `.env` file with the **ACCESS_CONTROL_ALLOW_X** properties in `.env.example`. You can also just let the config file use the default values.
2. Run the server: `go run cmd/mobile-security-service-server/main.go`
3. Make a request to the API using `curl`:

```sh
curl -H "Origin: http://example.com" \
  -H "Access-Control-Request-Method: GET" \
  -H "Access-Control-Request-Headers: X-Requested-With" \
  -X OPTIONS --verbose \
  http://localhost:3000 # Use the port your server is running on
```
 
4. Check the output:

```sh
< HTTP/1.1 204 No Content
< Access-Control-Allow-Headers: X-Requested-With
< Access-Control-Allow-Methods: GET
< Access-Control-Allow-Origin: *
```

## Checklist:


## Progress


## Additional Notes

If you set the `ACCESS_CONTROL_ALLOW_ORIGIN` to a value other than "*" it won't show in the output of the curl command in step 4. This will need to be verified when we have the web application created.
